File: Latihan/LatihanIdlingResource/java/app/src/main/java/com/dicoding/picodiploma/myidleresource/EspressoIdlingResource.java
package com.dicoding.picodiploma.myidleresource;

import androidx.test.espresso.IdlingResource;
import androidx.test.espresso.idling.CountingIdlingResource;

public class EspressoIdlingResource {
    private static final String RESOURCE = "GLOBAL";
    private static CountingIdlingResource espressoTestIdlingResource = new CountingIdlingResource(RESOURCE);

    static void increment() {
        espressoTestIdlingResource.increment();
    }

    static void decrement() {
        espressoTestIdlingResource.decrement();
    }

    static IdlingResource getEspressoIdlingResource() {
        return espressoTestIdlingResource;
    }
}

File: Latihan/LatihanIdlingResource/kotlin/app/src/main/java/com/dicoding/picodiploma/myidleresource/EspressoIdlingResource.kt
package com.dicoding.picodiploma.myidleresource

import androidx.test.espresso.IdlingResource
import androidx.test.espresso.idling.CountingIdlingResource

object EspressoIdlingResource {
    private val RESOURCE: String? = "GLOBAL"
    private val espressoTestIdlingResource = CountingIdlingResource(RESOURCE)
    fun increment() {
        espressoTestIdlingResource.increment()
    }

    fun decrement() {
        espressoTestIdlingResource.decrement()
    }

    fun getEspressoIdlingResource(): IdlingResource {
        return espressoTestIdlingResource
    }
}